core/trino-main/src/main/java/io/trino/sql/gen/columnar/CallColumnarFilterGenerator.java

```diff
@@ -0,0 +1,464 @@
+import io.trino.spi.function.ScalarFunctionImplementation;
+import io.trino.sql.gen.Binding;
+import io.trino.sql.gen.CallSiteBinder;
+import io.trino.sql.relational.CallExpression;
+import io.trino.sql.relational.ConstantExpression;
+import io.trino.sql.relational.InputReferenceExpression;
+import io.trino.sql.relational.RowExpression;
+import io.trino.type.FunctionType;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static io.airlift.bytecode.Access.FINAL;
+import static io.airlift.bytecode.Access.PUBLIC;
+import static io.airlift.bytecode.Access.a;
+import static io.airlift.bytecode.Parameter.arg;
+import static io.airlift.bytecode.ParameterizedType.type;
+import static io.airlift.bytecode.expression.BytecodeExpressions.add;
+import static io.airlift.bytecode.expression.BytecodeExpressions.constantInt;
+import static io.airlift.bytecode.expression.BytecodeExpressions.inlineIf;
+import static io.airlift.bytecode.expression.BytecodeExpressions.lessThan;
+import static io.airlift.bytecode.instruction.Constant.loadBoolean;
+import static io.airlift.bytecode.instruction.Constant.loadDouble;
+import static io.airlift.bytecode.instruction.Constant.loadLong;
+import static io.airlift.bytecode.instruction.Constant.loadString;
+import static io.trino.spi.function.InvocationConvention.InvocationArgumentConvention.BLOCK_POSITION;
+import static io.trino.spi.function.InvocationConvention.InvocationArgumentConvention.NEVER_NULL;
+import static io.trino.spi.function.InvocationConvention.InvocationReturnConvention.FAIL_ON_NULL;
+import static io.trino.spi.function.InvocationConvention.simpleConvention;
+import static io.trino.sql.gen.BytecodeUtils.invoke;
+import static io.trino.sql.gen.BytecodeUtils.loadConstant;
+import static io.trino.sql.gen.columnar.ColumnarFilterCompiler.createClassInstance;
+import static io.trino.sql.gen.columnar.ColumnarFilterCompiler.declareBlockVariables;
+import static io.trino.sql.gen.columnar.ColumnarFilterCompiler.generateBlockMayHaveNull;
+import static io.trino.sql.gen.columnar.ColumnarFilterCompiler.generateBlockPositionNotNull;
+import static io.trino.sql.gen.columnar.ColumnarFilterCompiler.generateConstructor;
+import static io.trino.sql.gen.columnar.ColumnarFilterCompiler.generateGetInputChannels;
+import static io.trino.util.CompilerUtils.makeClassName;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class CallColumnarFilterGenerator
+{
+    private final CallExpression callExpression;
+    private final FunctionManager functionManager;
+
+    public CallColumnarFilterGenerator(CallExpression callExpression, FunctionManager functionManager)
+    {
+        callExpression.getResolvedFunction().getSignature().getArgumentTypes().forEach(type -> {
+            if (type instanceof FunctionType) {
+                throw new UnsupportedOperationException(format("Function with argument type %s is not supported", type));
```

Comment:
I updated the error message, but kept the `UnsupportedOperationException` instead of `checkArgument` using `UnsupportedOperationException` helps to distinguish between unsupported cases and genuine bugs (I log unsupported cases at debug level and failures at warning level).